main.go

```diff
@@ -120,6 +121,24 @@ func main() {
 		startDebugServer(config)
 	}
 
+	// This goroutine is supposed to wake up every 20 milliseconds. If it
+	// finds itself slept for an unusually long time, it writes a log entry
+	// to record the delay. The delay could be due to GC pause, scheduler
+	// issue or others. This may help us investigate the 504 issue that we
+	// have seen recently.
+	go func() {
```

Comment:
What would this find?

Reply:
If we find in the log that 'All peers timed out' happens together with this 'unusual delay', we can infer there may be some system-wide issue. It could be a long GC pause, CPU being throttled, scheduler issue...?